src/status_im2/contexts/chat/images_horizontal/view.cljs

```diff
@@ -0,0 +1,93 @@
+(ns status-im2.contexts.chat.images-horizontal.view
+  (:require [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]
+            [react-native.fast-image :as fast-image]
+            [utils.re-frame :as rf]
+            [react-native.safe-area :as safe-area]
+            [reagent.core :as reagent]
+            [oops.core :as oops]
+            [status-im2.contexts.chat.images-horizontal.style :as style]
+            [utils.datetime :as datetime]))
+
+(defn image
+  [message]
+  (let [shared-element-id (rf/sub [:shared-element-id])
+        window-width      (:width (rn/get-window))
+        height            (* (:image-height message) (/ window-width (:image-width message)))]
+    [fast-image/fast-image
+     {:source    {:uri (:image (:content message))}
+      :style     {:width         window-width
+                  :height        height
+                  :border-radius 12}
+      :native-ID (when (= shared-element-id (:message-id message)) :shared-element)}]))
+
+(defn get-item-layout
+  [_ index]
+  (let [window-width (:width (rn/get-window))]
+    #js {:length window-width :offset (* window-width index) :index index}))
+
+(defn on-viewable-items-changed
+  [e]
+  (rf/dispatch [:chat.ui/update-shared-element-id
+                (:message-id (oops/oget (first (oops/oget e "changed")) "item"))]))
```

Comment:
@OmarBasem, `oops/oget` was designed on purpose to work well with the thread first macro, so you can *pipeline* calls, just as when using functions like `get` or `get-in`.

Suggestion:

```clojure
(-> e
    (oops/oget :changed)
    first
    (oops/oget :item))
```

Also notice we should be using keywords instead of strings to access properties. See proof below:

```clojure
(assert "banana"
        (-> (clj->js {"changed" [{"item" "banana"}]})
            (oget :changed)
            first
            (oget :item)))
```

Reply:
oh nice, didn't know you can use keyword